csharp/src/Ice/Internal/ConnectionFactory.cs

```diff
@@ -1329,133 +1329,125 @@ public override void message(ThreadPoolCurrent current)
+            //
+            // Now accept a new connection.
+            //
+            Transceiver transceiver = null;
+            try
+            {
+                transceiver = _acceptor.accept();
+
+                if (_maxConnections > 0 && _connections.Count == _maxConnections)
                 {
-                    transceiver = _acceptor.accept();
+                    // Can't accept more connections, so we abort this transport connection.
 
-                    if (_maxConnections > 0 && _connections.Count == _maxConnections)
+                    if (_instance.traceLevels().network >= 2)
                     {
-                        // Can't accept more connections, so we abort this transport connection.
-
-                        if (_instance.traceLevels().network >= 2)
-                        {
-                            _instance.initializationData().logger.trace(
-                                _instance.traceLevels().networkCat,
-                                $"rejecting new {_endpoint.protocol()} connection\n{transceiver}\nbecause the maximum number of connections has been reached");
-                        }
-
-                        try
-                        {
-                            transceiver.close();
-                        }
-                        catch
-                        {
-                            // Ignore
-                        }
-                        transceiver.destroy();
-                        return;
+                        _instance.initializationData().logger.trace(
+                            _instance.traceLevels().networkCat,
+                            $"rejecting new {_endpoint.protocol()} connection\n{transceiver}\nbecause the maximum number of connections has been reached");
                     }
 
-                    if (_instance.traceLevels().network >= 2)
+                    try
                     {
-                        StringBuilder s = new StringBuilder("trying to accept ");
-                        s.Append(_endpoint.protocol());
-                        s.Append(" connection\n");
-                        s.Append(transceiver.ToString());
-                        _instance.initializationData().logger.trace(_instance.traceLevels().networkCat, s.ToString());
+                        transceiver.close();
                     }
-                }
-                catch (Ice.SocketException ex)
-                {
-                    if (Network.noMoreFds(ex.InnerException))
+                    catch
                     {
-                        string s = "can't accept more connections:\n" + ex + '\n' + _acceptor.ToString();
-                        _instance.initializationData().logger.error(s);
-                        Debug.Assert(_acceptorStarted);
-                        _acceptorStarted = false;
-                        _adapter.getThreadPool().finish(this);
-                        closeAcceptor();
+                        // Ignore
```

Comment:
In many other places, we log as errors unexpected exceptions from the transport implementation.